tests/run-make/split-debuginfo/Makefile

```diff
@@ -11,23 +13,77 @@ off:
 	[ ! -d $(TMPDIR)/foo.dSYM ]
 
 # Packed by default, but only if debuginfo is requested
-packed:
-	rm -rf $(TMPDIR)/*.dSYM
-	$(RUSTC) foo.rs
-	[ ! -d $(TMPDIR)/foo.dSYM ]
-	rm -rf $(TMPDIR)/*.dSYM
-	$(RUSTC) foo.rs -g
+packed: packed-remapped-scope packed-remapped-wrong-scope
+
+# - Debuginfo in binary file
+# - `.o` deleted
+# - `.dSYM` present
+# - in binary, paths from `N_SO` (source files) and `N_OSO` (object files) shouldn be remapped
+packed-remapped-scope:
+	$(RUSTC) $(UNSTABLEOPTS) -C split-debuginfo=packed -C debuginfo=2 \
+		--remap-path-prefix $(TMPDIR)=/a \
+		--remap-path-prefix $(HERE)=/b \
+		-Z remap-path-scope=object foo.rs -g
+	ls $(TMPDIR)/*.o && exit 1 || exit 0
 	[ -d $(TMPDIR)/foo.dSYM ]
-	rm -rf $(TMPDIR)/*.dSYM
-	$(RUSTC) foo.rs -g -C split-debuginfo=packed
+	# As of 2023-12, `OSO` should be the only thing that cannot be trimmed. See rust-lang/rust#116948
+	dsymutil -s $(TMPDIR)/foo | grep $(TMPDIR) || exit 1 # expected: `grep $(TMPDIR)` to exit 1
+	dsymutil -s $(TMPDIR)/foo | (! grep $(HERE)) || exit 1
+	rm -rf $(TMPDIR)/foo.dSYM
+	rm $(TMPDIR)/$(call BIN,foo)
```

Comment:
I'd suggest changing all of the remapping tests in this file to do a sanity check:
- remap paths to something that is unlikely to be contained in the binary by accident (e.g. `/my-remapped-src` and `/my-remapped-cwd` instead of `/a` and `/b`)
- check that `/my-remapped-src` and `/my-remapped-cwd` *are* contained in the various files, so that we make sure the tests would actually fail if something is wrong. These run-make tests are very prone to bugs where the tests silently pass without testing anything.

Reply:
Updated, but I don't think it is in a good state to merge.


If you look closer at the changes in tests in the last commit, you can see it is like a whack-a-mole game.

* Some symbols in split debuginfo file got overly remapped in `object` scope, and some should be remapped but didn't.
* So do unpacked debuginfo in `.o` files.
* `.debug-line` doesn't seem to be handled well.
* When unpacked `.o` object files contain debuginfo, do we consider them split debuginfo files?


That leads to a question: Does llvm provide such a granular API for rustc to implement RFC 3127? 

(or maybe this is not a task for me 😞)

Reply:
I haven't taken an in-depth look but [DIBuilder::createFile](https://llvm.org/doxygen/classllvm_1_1DIBuilder.html#afac26b4f6298753fa1566e66758c9dbe) does not seem to offer anything.

I think it's time to re-evaluate this particular aspect of RFC 3127. Maybe we just need to remove the split-/unsplit-debuginfo distinction and just have a single `debuginfo` scope. That still seems practical to me. I've created a todo for me to post something to the tracking issue. Let's mark the PR as blocked for now.

Reply:
Thanks for the review, Michael!

What I am worried about is that scope `object` and `debug` never actually work as defined in RFC 3127 unless we have some post-processing approaches. Reading through [this post](https://reproducible-builds.org/docs/build-path/) and other tools like clang and gcc, seems none of them offers such an option to deal with split-debuginfo files and binaries separately like RFC 3127 suggesting.